### PR TITLE
move prometheus register to pkg/endpoints/metrics/metrics.go

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**What this PR does / why we need it**:
delete  k8s.io/apiserver/pkg/endpoints/apiserver.go
move prometheus init from k8s.io/apiserver/pkg/endpoints/apiserver.go to k8s.io/apiserver/pkg/endpoints/metrics/metrics.go

Since  k8s.io/apiserver/pkg/endpoints/apiserver.go only has a metrics init func, which should be in metrics pkg.

**Release note**:

```release-note
NONE
```

/assign @sttts @liggitt

### CODE REVIEW

**Comment by hzxuzhonghu** [comment]
/retest

azure ut data race

File: staging/src/k8s.io/apiserver/pkg/endpoints/metrics/metrics.go [hunk 1]
```diff
@@ -81,8 +81,8 @@ var (
 	kubectlExeRegexp = regexp.MustCompile(`^.*((?i:kubectl\.exe))`)
 )
 
-// Register all metrics.
-func Register() {
+func init() {
```
Comment (liggitt) [inline staging/src/k8s.io/apiserver/pkg/endpoints/metrics/metrics.go]: not a fan of importing the metrics package autoregistering

Comment (hzxuzhonghu) [reply staging/src/k8s.io/apiserver/pkg/endpoints/metrics/metrics.go]: Previously, autoregistering by importing `k8s.io/apiserver/pkg/endpoints`

Comment (sttts) [reply staging/src/k8s.io/apiserver/pkg/endpoints/metrics/metrics.go]: Is there an easy way to lazily register when metrics are actually used?

Comment (hzxuzhonghu) [reply staging/src/k8s.io/apiserver/pkg/endpoints/metrics/metrics.go]: Yes, we can implement lazy register by register in `Record` and `Record`. But why we need  `lazily register` ?

Comment (sttts) [reply staging/src/k8s.io/apiserver/pkg/endpoints/metrics/metrics.go]: I agree. pkg/endpoints/metrics is pretty deep in the apiserver. Anybody importing that outside of k8s.io/apiserver?

Comment (hzxuzhonghu) [reply staging/src/k8s.io/apiserver/pkg/endpoints/metrics/metrics.go]: I donot know.  Maybe user private extended apiservers import it.

Comment (hzxuzhonghu) [reply staging/src/k8s.io/apiserver/pkg/endpoints/metrics/metrics.go]: But these are not directly import.

**Pull request merged**